addons/purchase_requisition/models/purchase.py

```diff
@@ -101,26 +151,135 @@ def create(self, vals_list):
 
+    def action_create_alternative(self):
+        ctx = self.env.context.copy()
+        ctx.update({
+            'default_origin_po_id': self.id,
+        })
+        return {
+            'name': _('Create alternative'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'purchase.requisition.create.alternative',
+            'view_id': self.env.ref('purchase_requisition.purchase_requisition_create_alternative_form').id,
+            'target': 'new',
+            'context': ctx,
+        }
+
+    def action_compare_alternative_lines(self):
+        best_price_ids, best_date_ids = self.get_tender_best_date_and_price_lines()
+        ctx = self._context.copy()
+        ctx.update({
+            'search_default_groupby_product': True,
+            'params': {
+                'best_price_ids': best_price_ids,
+                'best_date_ids': best_date_ids,
+            },
+        })
+        view_id = self.env.ref('purchase_requisition.purchase_order_line_compare_tree').id
+        return {
+            'name': _('Compare Order Lines'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'list',
+            'res_model': 'purchase.order.line',
+            'views': [(view_id, "list")],
+            'domain': [('order_id', 'in', self.ids + self.alternative_po_ids.ids)],
+            'context': ctx,
+        }
+
+    def get_tender_best_date_and_price_lines(self):
+        product_to_best_price_line = defaultdict(lambda: self.env['purchase.order.line'])
+        product_to_best_date_line = defaultdict(lambda: self.env['purchase.order.line'])
+        order_lines = self.order_line | self.alternative_po_ids.order_line
+        for line in order_lines:
+            if not line.product_qty or not line.price_subtotal or line.state in ['cancel', 'purchase', 'done']:
+                continue
+            if not product_to_best_price_line[line.product_id] or product_to_best_price_line[line.product_id][0].price_subtotal > line.price_subtotal:
+                product_to_best_price_line[line.product_id] = line
+            elif product_to_best_price_line[line.product_id][0].price_subtotal == line.price_subtotal:
+                product_to_best_price_line[line.product_id] |= line
+            if not product_to_best_date_line[line.product_id] or product_to_best_date_line[line.product_id][0].date_planned > line.date_planned:
+                product_to_best_date_line[line.product_id] = line
+            elif product_to_best_date_line[line.product_id][0].date_planned == line.date_planned:
+                product_to_best_date_line[line.product_id] |= line
+        best_price_ids = set()
+        best_date_ids = set()
+        for lines in product_to_best_price_line.values():
+            best_price_ids.update(lines.ids)
+        for lines in product_to_best_date_line.values():
+            best_date_ids.update(lines.ids)
+        return best_price_ids, best_date_ids
```

Comment:
I think it's better to parse them into lists before to return them. That way, you'll get an array with actual numbers instead of a string in the JS (`set` and `dict` are passed as string :shrug: )
```suggestion
        return list(best_price_ids), list(best_date_ids)
```